### PR TITLE
Rails upgrade merged changes

### PR DESCRIPTION
**Pull request by rbclark**
This PR contains all changes so far done for Rails 5.1 and has also already been rebased against the cqm-execution-service changes.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR: https://jira.mitre.org/browse/CYPRESS-347
- [x] Internal ticket links to this PR
- [x] Code diff has been done and been reviewed
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass

**Reviewer 1:**

Name: Dave Czulada
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**Reviewer 2:**

Name: @okeefm 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

### CODE REVIEW

**Comment by rbclark** [comment]
Finally passing travis!

File: lib/cypress/patient_zipper.rb [hunk 1]
```diff
@@ -7,24 +7,25 @@
...
       # TODO: R2P: make sure patient export works with HDS HTML exporter
-      hdsrecord = @qdm_patient_converter.to_hds(patient)
-      hdsrecord.bundle_id = patient.bundleId
-      # replace the end_time with nil if the end_time is > 1893474001, which is 1/1/2030.
-      # This will get the HTML exporter to export the end time as present.
-      hdsrecord.entries.each { |de| de.end_time = nil if de.end_time && de.end_time > 1_893_474_001 }
-      EXPORTER.export(hdsrecord, measures)
+      # hdsrecord = @qdm_patient_converter.to_hds(patient)
+      # hdsrecord.bundle_id = patient.bundleId
+      # EXPORTER.export(hdsrecord, measures)
```
Comment (dczulada) [inline lib/cypress/patient_zipper.rb]: Commented out code (and TODOs) can be removed.

File: Gemfile [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
-gem 'cqm-models', git: 'https://github.com/projecttacoma/cqm-models.git', branch: 'qdm5.3'
-gem 'cqm-parsers', '~> 0.2.1'
-gem 'cqm-validators', '~> 0.1.0'
-gem 'health-data-standards', '~> 4.3.0'
-
-gem 'quality-measure-engine',
-    git: 'https://github.com/projectcypress/quality-measure-engine.git', branch: 'bump_mongoid'
+gem 'cqm-models', git: 'https://github.com/projecttacoma/cqm-models.git', branch: 'remove_hds'
+# gem 'cqm-parsers', git: 'https://github.com/projecttacoma/cqm-parsers.git', branch: 'master'
+gem 'cql_qdm_patientapi', git: 'https://github.com/projecttacoma/cql_qdm_patientapi.git', branch: 'rails_upgrade'
```
Comment (dczulada) [inline Gemfile]: Is cql_qdm_patientapi actually needed by cypress?

File: lib/ext/string.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class String
+  def to_boolean
+    %w[1 true t].include?(downcase)
```
[the remark this answers is not in the document]
Comment (rbclark) [reply lib/ext/string.rb]: In this case since it is an override of the ruby String variable, I think downcase is just calling the downcase method on the string object.

File: lib/ext/value_set.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+Concept = QDM::Concept
+
+class ValueSet
+  include Mongoid::Document
+  store_in collection: 'health_data_standards_svs_value_sets'
+
+  belongs_to :bundle, :class_name => 'Bundle', :inverse_of => :value_sets
+end
+
+class Concept
```
[the remark this answers is not in the document]
Comment (okeefm) [reply lib/ext/value_set.rb]: Update: per @dczulada these are only here as these fields are still coming in from Bonnie-created Bundles. Might be worth adding a `TODO:` to remove this when a new bundle no longer includes those fields.

File: app/views/records/show.html.erb [hunk 5]
```diff
@@ -1,4 +1,4 @@
-<h1>Patient Information for <%= full_name(@record) %></h1>
+<!--h1>Patient Information for <%= full_name(@record) %></h1-->
```
[the remark this answers is not in the document]
Comment (dczulada) [reply app/views/records/show.html.erb]: good catch, we should probably remove that commented out code as well.  That was removed because it is not redundant.

Old view
image

New View
image

File: app/controllers/admin/settings_controller.rb [hunk 6]
```diff
@@ -12,7 +12,7 @@ def edit
     def update
       update_application_mode params[:mode], params[:custom_options]
       # Grab the parameters we are able to update directly and throw them to the settings model update method
-      update_settings = params.select { |key, _| key.match(/website|mailer|banner|message/) }
+      # update_settings = params.select { |key, _| key.match(/website|mailer|banner|message/) }
```
[the remark this answers is not in the document]
Comment (rbclark) [reply app/controllers/admin/settings_controller.rb]: Fixed

File: lib/cypress/patient_zipper.rb (hunk 1 above)
Comment (rbclark) [reply lib/cypress/patient_zipper.rb]: Fixed

File: app/views/records/show.html.erb (hunk 5 above)
[the remark this answers is not in the document]
Comment (rbclark) [reply app/views/records/show.html.erb]: Removed
